master/internal/resourcemanagers/priority.go

```diff
@@ -244,8 +245,8 @@ func sortTasksByPriorityAndTimestamp(
 	priorityToPendingTasksMap := make(map[int][]*sproto.AllocateRequest)
 	priorityToScheduledTaskMap := make(map[int][]*sproto.AllocateRequest)
 
-	for it := taskList.iterator(); it.next(); {
-		req := it.value()
+	reqs := sortTasks(taskList, groups, false)
```

Comment:
i dont think we need this change if we also change the comparator passed to treeset.NewWith, so that the iterator is still in the correct order? if so, why this change? after a quick glance, that looks more correct for elsewhere the iterator is used (where it matters, and order is important).

Reply:
yeah we shouldn't need to but I wanted to make sure there is less chance of deviation on how the jobs are sorted but I can take it out for this particular fix.

Reply:
we could pass in the filter to  sortTasks too if that's a concern.